controllers/drpolicy_controller.go

```diff
@@ -90,6 +97,67 @@ func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{}, nil
 }
 
+func validate(ctx context.Context, drpolicy *ramen.DRPolicy, apiReader client.Reader,
+	client client.Client, objectStoreGetter ObjectStoreGetter, log logr.Logger,
+) error {
+	var (
+		conditionSetTrue  func(reason, message string) error
+		conditionSetFalse func(reason string, err error) error
+	)
+
+	if condition := util.DrpolicyValidatedConditionGet(drpolicy); condition != nil {
+		if condition.Status == metav1.ConditionTrue {
```

Comment:
(future) If ramen config changed (which is not detected or reconciled, hence assume it was done async). We will not rerun the validation. Should we consider running this every time, or at least when the generation of the condition and the resource are different?

Reply:
I agree it should be possible to transition from valid to invalid if say an s3 profile stored in ramen config, or the secret that is points to changes.  It seems the right way to do that is by drpolicy watching for changes to a cluster's ramen config and s3 secrets referenced by an s3 profile referenced in the drpolicy.

Or it could become invalid if a drpolicy changed to point to a different s3 profile, which would result in a drpolicy reconcile call, but not a revalidation.  At this point I understand drpolicy modification is not supported.  For example, removing clusters could result in managed cluster vrg clusterroles being leaked.

However, say s3 access permission is revoked for the user specified in an s3 profile's secret.  This seems to be a way for the s3 store connectivity to become invalid without a change to any kubernetes resources and hence no normal way for drpolicy reconciler to be run.

But I do think the point is to validate a drpolicy's s3 profiles and so previously valid ones should be re-validated on any changes to them that drpolicy can be made aware of by normal means.

A workaround to revalidate is to delete and recreate a drpolicy, or if it is mutable, delete its validated condition or modify its validated condition's state to false or unknown which are both considered "invalid".